src/operators/collection_operator.rs

```diff
@@ -133,20 +135,57 @@ pub fn create_card_bookmark_query(
 
 pub fn get_bookmarks_for_collection_query(
     collection: uuid::Uuid,
+    current_user_id: Option<uuid::Uuid>,
     pool: web::Data<Pool>,
-) -> Result<Vec<CardCollectionBookmark>, DefaultError> {
-    use crate::data::schema::card_collection_bookmarks::dsl::*;
+) -> Result<Vec<CardMetadataWithVotes>, DefaultError> {
+    use crate::data::schema::card_collection_bookmarks::dsl as card_collection_bookmarks_columns;
+    use crate::data::schema::card_metadata::dsl as card_metadata_columns;
 
     let mut conn = pool.get().unwrap();
 
-    let bookmarks = card_collection_bookmarks
-        .filter(collection_id.eq(collection))
+    let bookmarks = card_collection_bookmarks_columns::card_collection_bookmarks
+        .filter(card_collection_bookmarks_columns::collection_id.eq(collection))
         .load::<CardCollectionBookmark>(&mut conn)
         .map_err(|_err| DefaultError {
             message: "Error getting bookmarks",
         })?;
 
-    Ok(bookmarks)
+    let bookmark_metadata = card_metadata_columns::card_metadata
+        .filter(
+            card_metadata_columns::id.eq_any(
+                bookmarks
+                    .iter()
+                    .map(|bookmark| bookmark.card_metadata_id)
+                    .collect::<Vec<uuid::Uuid>>(),
+            ),
+        )
+        .select((
+            card_metadata_columns::id,
+            card_metadata_columns::content,
+            card_metadata_columns::link,
+            card_metadata_columns::author_id,
+            card_metadata_columns::qdrant_point_id,
+            card_metadata_columns::created_at,
+            card_metadata_columns::updated_at,
+            card_metadata_columns::oc_file_path,
+            card_metadata_columns::card_html,
+            card_metadata_columns::private,
+        ))
```

Comment:
Yeah the issue is that there is a tsvector column in the pg database for the full text search which isn't representable in rust trivially so I just put all the other columns in so that I don't need to deal with that